core/src/operations/decorators/advice.rs

```diff
@@ -270,7 +316,60 @@ pub enum AdviceInjector {
     ///
     /// Where PK is the public key corresponding to the signing key, MSG is the message, SIG_DATA
     /// is the signature data.
-    SigToStack { kind: SignatureKind },
+    FalconSigToStack,
+}
+
+impl AdviceInjector {
```

Comment:
Probably not for this PR, but I'm wonder if we actually need the `AdviceInjector` enum at all. I think pretty much in all cases it can be replaced with `Operation::Emit(EVENT_ID)` and the code would be just as readable.